parachain/pallets/ethereum2-light-client/src/lib.rs

```diff
@@ -0,0 +1,402 @@
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+use codec::{Decode, Encode};
+use frame_support::{dispatch::DispatchResult, log, traits::Get, transactional};
+use frame_system::ensure_signed;
+use scale_info::TypeInfo;
+use sp_core::H256;
+use sp_runtime::RuntimeDebug;
+use sp_std::prelude::*;
+use std::cmp;
+
+pub use snowbridge_ethereum::Header as EthereumHeader;
+
+/// https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/sync-protocol.md#misc
+/// The minimum number of validators that needs to sign update
+const MIN_SYNC_COMMITTEE_PARTICIPANTS: u8 = 1;
+
+const EPOCHS_PER_SYNC_COMMITTEE_PERIOD: u32 = 512;
+
+const SECONDS_PER_SLOT: u32 = 12;
+
+const SLOTS_PER_EPOCH: u32 = 32;
+
+const SYNC_COMMITTEE_SIZE: u32 = 256;
+
+const UPDATE_TIMEOUT: u32 = SLOTS_PER_EPOCH * EPOCHS_PER_SYNC_COMMITTEE_PERIOD;
+
+/// Beacon block header as it is stored in the runtime storage.
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+// https://yeeth.github.io/BeaconChain.swift/Structs/BeaconBlockHeader.html#/s:11BeaconChain0A11BlockHeaderV9signature10Foundation4DataVvp
+pub struct BeaconBlockHeader {
+	// The slot for which this block is created. Must be greater than the slot of the block defined by parentRoot.
+	pub slot: u64,
+	// The block root of the parent block, forming a block chain.
+	pub parent_root: H256,
+	// The hash root of the post state of running the state transition through this block.
+	pub state_root: H256,
+	// BLS Signature of the block by the block proposer, TODO type isn't right yet
+	pub signature: String,
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct Root {
+	// TODO: Add
+}
+
+/// Sync committee as it is stored in the runtime storage.
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct SyncCommittee {
+	// TODO: Add
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct SyncAggregate {
+	pub sync_committee_bits: u64, // TODO is this right even
+}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct Version {}
+
+#[derive(Clone, Default, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
+pub struct LightClientUpdate {
+	/// The beacon block header that is attested to by the sync committee
+	pub attested_header: BeaconBlockHeader,
+	///  Next sync committee corresponding to the active header
+	pub next_sync_committee: SyncCommittee,
+	/// Vector[Bytes32, floorlog2(NEXT_SYNC_COMMITTEE_INDEX)]
+	pub next_sync_committee_branch: Vec<H256>,
+	/// The finalized beacon block header attested to by Merkle branch
+	pub finalized_header: Option<BeaconBlockHeader>,
+	/// Vector[Bytes32, floorlog2(FINALIZED_ROOT_INDEX)]
+	pub finality_branch: Vec<H256>,
+	///  Sync committee aggregate signature
+	pub sync_aggregate: SyncAggregate,
+	///  Fork version for the aggregate signature
+	pub pubfork_version: Version,
+}
+
+pub use frame_system::pallet::*;
```

Comment:
You're importing frame_system's pallet contents instead of the contents of this pallet